Tween/ParalleledAnimation.cs

```diff
@@ -13,13 +13,17 @@ protected override void PlayQueuedAnimations()
 
 			if (Animations.Count > 0)
 			{
-				Animations.ForEach(animation => animation.Play(HandleEachAnimationCompleted, () =>
+				for (var i = 0; i < Animations.Count; ++i)
 				{
-					if (!animation.IsValid)
+					var animation = Animations[i];
+					animation.Play(HandleEachAnimationCompleted, () =>
 					{
-						Animations.Remove(animation);
-					}
-				}));
+						if (IsPlaying && !animation.IsValid)
```

Comment:
Can animations not be paused? Or delayed? Wouldn't this line cause those to be discarded without playing?

Reply:
If `IsPuased == true` then `IsPlaying` must be true. Otherwise, it will throw.

We don't support delayed animation. (delay animation -- `WaitAnimation` is just an animation with a timer in)